public/app/features/explore/Graph.tsx

```diff
@@ -133,35 +133,29 @@ export class Graph extends PureComponent<GraphProps, GraphState> {
   }
 
   onPlotSelected = (event, ranges) => {
-    if (this.props.onChangeTime) {
+    const { onChangeTime, timeZone } = this.props;
+    if (onChangeTime) {
       const range = {
-        from: moment(ranges.xaxis.from),
-        to: moment(ranges.xaxis.to),
+        from: timeZone.isUtc ? moment.utc(ranges.xaxis.from) : moment(ranges.xaxis.from),
+        to: timeZone.isUtc ? moment.utc(ranges.xaxis.to) : moment(ranges.xaxis.to),
       };
       this.props.onChangeTime(range);
     }
   };
 
   getDynamicOptions() {
-    const { range, width } = this.props;
+    const { range, width, timeZone } = this.props;
     const ticks = (width || 0) / 100;
-    let { from, to } = range;
-    if (!moment.isMoment(from)) {
-      from = dateMath.parse(from, false);
-    }
-    if (!moment.isMoment(to)) {
-      to = dateMath.parse(to, true);
-    }
-    const min = from.valueOf();
-    const max = to.valueOf();
+    const min = range.from.valueOf();
```

Comment:
Related to comment regarding `onPlotSelected` above, maybe this component should have numeric from/to props to represent epoch ms and GraphContainer would be responsible converting to epoch ms. This way we remove moment dependency from this component.